src/tests/ftest/deployment/target_failure.py

```diff
@@ -0,0 +1,417 @@
+
+        Args:
+            container (TestContainer): Container to call get-prop.
+            expected_health (str): Expected value in the Health field.
+
+        Returns:
+            bool: True if expected_health matches the one obtained from get-prop.
+
+        """
+        actual_health = None
+
+        output = container.get_prop()
+        for cont_prop in output["response"]:
+            if cont_prop["name"] == "status":
+                actual_health = cont_prop["value"]
+                break
+
+        return actual_health == expected_health
+
+    def measure_rebuild_time(self, operation, pool):
+        """Measure rebuild time.
+
+        This method is mainly for debugging purpose. We'll analyze the output when we
+        realize that rebuild is taking too long.
+
+        Args:
+            operation (str): Type of operation to print in the log.
+            pool (TestPool): Pool to call wait_for_rebuild().
+        """
+        start = float(time.time())
+        pool.wait_for_rebuild(to_start=True)
+        pool.wait_for_rebuild(to_start=False)
+        duration = float(time.time()) - start
+        self.log.info("%s duration: %.1f sec", operation, duration)
+
+    def test_target_failure_wo_rf(self):
+        """Jira ID: DAOS-10001.
+
+        Verify that failing (excluding) one target would cause an error to the ongoing
+        IOR. Also verify that reintegrating the excluded target will bring back the system
+        to the usable state.
+
+        1. Create a pool and a container.
+        2. Run IOR with oclass SX.
+        3. Exclude one target while IOR is running.
+        4. Verify the IOR failed.
+        5. Verify that the container's Health property is UNCLEAN.
+        6. Reintegrate the excluded target.
+        7. Verify that the container's Health property is HEALTHY.
+        8. Run IOR again.
+        9. Verify that there's no error this time.
+
+        :avocado: tags=all,full_regression
+        :avocado: tags=hw,medium,ib2
+        :avocado: tags=deployment,target_failure
+        :avocado: tags=target_failure_wo_rf
+        """
+        # 1. Create a pool and a container.
+        self.add_pool(namespace="/run/pool_size_ratio/*")
+        self.add_container(pool=self.pool, namespace="/run/container_wo_rf/*")
+
+        # 2. Run IOR with oclass SX so that excluding one target will result in a failure.
+        ior_results = {}
+        job_num = 1
+        oclass = self.params.get("oclass", "/run/object_class/wo_rf/*")
+        job = threading.Thread(
+            target=self.run_ior_report_error,
+            args=[ior_results, job_num, "test_file_1", oclass, self.pool, self.container])
+
+        job.start()
+
+        # Wait for a few seconds for IOR to start.
+        self.log.info("Waiting 5 sec for IOR to start writing data...")
+        time.sleep(5)
+
+        errors = []
+
+        # 3. Exclude one target while IOR is running.
+        self.pool.exclude(ranks=[1], tgt_idx="0")
```

Comment:
Sorry, there was typo in the comment. We're using SX object class, so the IO is supposed to be going into all the targets. The assumption is that the IOR is running when this command is called because we're running enough data that takes a while and we have waited 5 sec after the start of the IOR.